src/clib/pio_file.c

```diff
@@ -452,9 +451,9 @@ int PIOc_deletefile(int iosysid, const char *filename)
 
     /* Broadcast and check the return code. */
     if ((mpierr = MPI_Bcast(&ierr, 1, MPI_INT, ios->ioroot, ios->my_comm)))
-        return check_mpi(file, mpierr2, __FILE__, __LINE__);        
+        return check_mpi(NULL, mpierr2, __FILE__, __LINE__);        
     if (ierr)
-        return check_netcdf(file, ierr, __FILE__, __LINE__);
+        return check_netcdf(NULL, ierr, __FILE__, __LINE__);
```

Comment:
The failure is "invalid memory reference" caused by dereferencing a NULL pointer. A builder configured as optimize (assertions are disabled) will also fail.